### PR TITLE
Added search tab for communitiyes (fixes #88)

### LINKED ISSUE #88: Global Community Search
Possibly a tie-in with https://browse.feddit.de/

### PR DESCRIPTION
**Pull request by JakeShirley**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [x] This PR addresses one or more open issues that were assigned to me:
      - Fixed #88 
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

## About this Pull Request
Adds a new search tab to the bottom bar.

I implemented paging on my own because I didn't think the feed tracker stuff would scale to multiple result types at the moment.  It's.... not pretty.

A few other side bits:
- Added utility to `Int` to print things like `1.2k` and `1.1m`
- Updated search tracker to use views so we can display how many subscribers communities have
- This does NOT touch the existing search, but it should be downgrade to swapping between "Subscribed", "Local", and "All" (no search)

## Screenshots and Videos
image

image

## Additional Context
**Any additional context you'd like to add to help us review this PR**

### CODE REVIEW

File: Mlem/Models/Trackers/RecentSearchesTracker.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    func loadFromDisk() {
+        if FileManager.default.fileExists(atPath: AppConstants.recentSearchesFilePath.path) {
+            print("Favorite communities file exists, will attempt to load favorite communities")
+            do {
+                recentSearches = try decodeFromFile(
+                    fromURL: AppConstants.recentSearchesFilePath,
+                    whatToDecode: .recentSearches
+                ) as? [String] ?? []
+            } catch let decodingError {
+                print("Failed while decoding recent searches, erasing file: \(decodingError)")
```
Comment (EricBAndrews) [inline Mlem/Models/Trackers/RecentSearchesTracker.swift:23]: Does the file actually get erased?

File: Mlem/Views/Tabs/Search/Search View.swift [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    var body: some View {
+        NavigationStack(path: $navigationPath) {
+            content
+                .handleLemmyViews()
+                .navigationBarTitleDisplayMode(.inline)
+                .navigationTitle("Search")
+        }
+        .searchable(text: getSearchTextBinding(), prompt: "Search for communities")
+        .onSubmit(of: .search) {
+            preformSearch()
```
Comment (EricBAndrews) [inline Mlem/Views/Tabs/Search/Search View.swift:43]: nit:

preform -> perform

File: Mlem/Views/Tabs/Search/Search View.swift [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                                    if hasMorePages {
+                                        preformSearch()
+                                    }
+                                }
+                            }
+                        }
+                }
+            }
+        }
+    }
```
Comment (EricBAndrews) [inline Mlem/Views/Tabs/Search/Search View.swift:108]: Good for now, long-term we can look at maybe using one of those shiny generic feed trackers

File: Mlem/Views/Tabs/Search/Search View.swift [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    
+    private func loadRecentSearchesFromDisk() {
+        
+    }
```
Comment (EricBAndrews) [inline Mlem/Views/Tabs/Search/Search View.swift:112]: ?

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Couple minor things, but looks good overall!

File: Mlem/ContentView.swift [hunk 5]
```diff
@@ -37,13 +37,19 @@ struct ContentView: View {
                     Label(computeUsername(account: currentActiveAccount), systemImage: "person.circle")
                         .environment(\.symbolVariants, tabSelection == 3 ? .fill : .none)
                 }.tag(3)
+                
+                NavigationView {
+                    SearchView(account: currentActiveAccount)
+                } .tabItem {
+                    Label("Search", systemImage: "text.magnifyingglass")
```
Comment (EricBAndrews) [inline Mlem/ContentView.swift:44]: Can we make this `magnifyingglass` when it's not selected and `text.magnifyingglass` when it is?

**Comment by EricBAndrews** [comment]
> Design notes/questions:
> 
> Can we center the “clear recent searches” within that button?
> 
> Should we take it out of the button itself and just have it be text on the black background? Might be a bit cleaner that way? Thoughts?
> 
> Should the number of subscribers text remain white? Or would a gray look a little better?

I think the communities should be white and the subscribers gray--right now the weight feels backwards

**Review by WestonHanners — APPROVED** [review]

File: Mlem/Views/Tabs/Search/Search View.swift (hunk 4 above)
Comment (JakeShirley) [reply Mlem/Views/Tabs/Search/Search View.swift:112]: What? You don't like functions that do nothing and are never called??

**Comment by JakeShirley** [comment]
Addressed all of the feedback other than the label coloring of names / subs.  I think we can do this but that will be a little deeper surgery for the community link view and it'd be nice to get a beta out this morning.

**Review by EricBAndrews — APPROVED** [review]

**Pull request merged**